Re-Import html/canvas WPT

karlcow opened the pull request:
#### fa36460f1e5155528a609d22f3e98d332f11cffe
<pre>
Re-Import html/canvas WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=274304">https://bugs.webkit.org/show_bug.cgi?id=274304</a>
<a href="https://rdar.apple.com/128259734">rdar://128259734</a>

re-importing web-platform tests for html/canvas

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/307a2512c8b20feb1641ebb6c4ef21efc3bacda0">https://github.com/web-platform-tests/wpt/commit/307a2512c8b20feb1641ebb6c4ef21efc3bacda0</a>

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/html/canvas/element/*
* LayoutTests/imported/w3c/web-platform-tests/html/canvas/historical.*
* LayoutTests/imported/w3c/web-platform-tests/html/canvas/offscreen/*
* LayoutTests/imported/w3c/web-platform-tests/html/canvas/tools/*
  Re-import of hundred of tests and tools for html/canvas
  Removing some expectations, Adding new ones.
* LayoutTests/imported/w3c/web-platform-tests/html/canvas/tools/gentestutilsunion.py:
(_Variant._render_param):
(_Variant.generate_expected_image):
(_VariantGrid.merge_params):
* LayoutTests/TestExpectations:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fa36460f1e5155528a609d22f3e98d332f11cffe

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67004 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46379 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19626 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71039 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18137 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54178 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17920 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53736 "layout-tests (failure)") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12204 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70071 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42656 "Exiting early after 10 failures. 2992 tests run. 1 failures") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57971 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34256 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39327 "Found 1 new test failure: imported/w3c/web-platform-tests/html/canvas/element/manual/compositing/colr-glyph-composition.html (failure)") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15365 "Exiting early after 11 failures. 2116 tests run. 4 flakes 3 failures") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16491 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61248 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15706 "Found 1 new test failure: imported/w3c/web-platform-tests/html/canvas/element/manual/compositing/colr-glyph-composition.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72740 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10961 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15042 "Found 1 new test failure: imported/w3c/web-platform-tests/html/canvas/element/manual/compositing/colr-glyph-composition.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61211 "layout-tests (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10993 "Built successfully") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58030 "Exiting early after 10 failures. 313 tests run. 1 failures") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61286 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8999 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2606 "Found 1 new test failure: imported/w3c/web-platform-tests/html/canvas/element/manual/compositing/colr-glyph-composition.html (failure)") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42186 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43263 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44446 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43006 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Can you remove the duplicate title from the commit message?

karlcow committed:
Re-Import html/canvas WPT
https://bugs.webkit.org/show_bug.cgi?id=274304
rdar://128259734

re-importing web-platform tests for html/canvas

Upstream commit: https://github.com/web-platform-tests/wpt/commit/307a2512c8b20feb1641ebb6c4ef21efc3bacda0

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/html/canvas/element/*
* LayoutTests/imported/w3c/web-platform-tests/html/canvas/historical.*
* LayoutTests/imported/w3c/web-platform-tests/html/canvas/offscreen/*
* LayoutTests/imported/w3c/web-platform-tests/html/canvas/tools/*
  Re-import of hundred of tests and tools for html/canvas
  Removing some expectations, Adding new ones.
* LayoutTests/imported/w3c/web-platform-tests/html/canvas/tools/gentestutilsunion.py:
(_Variant._render_param):
(_Variant.generate_expected_image):
(_VariantGrid.merge_params):
* LayoutTests/TestExpectations:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
